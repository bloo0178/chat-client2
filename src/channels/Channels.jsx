import React from "react";
import { withStyles } from "@material-ui/core/styles";
import ChannelList from "./ChannelList";
import CreateChannelButton from "./CreateChannelButton";

const styles = {
  root: {
    width: "100%",
    display: "flex",
    direction: "column",
    justifyContent: "center",
    alignItems: "center"
  },
  channelList: {
    width: "75%",
    minWidth: 300,
    maxWidth: 600,
    padding: "3rem",
    justifyContent: "center"
  },
  createChannel: {
    position: "absolute",
    bottom: 0,
    right: 0,
    margin: 15
  }
};

const Channels = props => {
  const {
    classes: { root, createChannel, channelList },
    history
  } = props;
  return (
    <React.Fragment>
      <div>Navbar w/ history</div>
      <div className={root}>
        <div className={channelList}>
          <ChannelList history={history} />
        </div>
        <div className={createChannel}>
          <CreateChannelButton history={history} />
        </div>
      </div>
    </React.Fragment>
  );
};

export default withStyles(styles)(Channels);
