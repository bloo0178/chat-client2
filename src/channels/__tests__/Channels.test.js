import React from 'react';
import {shallow} from 'enzyme';
import Channels from '../Channels';

describe("<Channels />", () => {
    let wrapper;
    const props = {

    }
    beforeEach(() => {
        wrapper = shallow(<Channels {...props} />).dive();
    })

    test("does it render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })
})