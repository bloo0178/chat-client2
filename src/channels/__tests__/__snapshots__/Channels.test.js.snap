// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Channels /> does it render correctly 1`] = `
<Fragment>
  <WithStyles(Navbar) />
  <div
    className="Channels-channelsContainer-1"
  >
    <div
      className="Channels-channelList-2"
    >
      <WithStyles(ChannelList)
        channels={Array []}
        enterChannel={[Function]}
      />
    </div>
    <div
      className="Channels-createChannel-3"
    >
      <CreateChannelButton
        enterChannel={[Function]}
      />
    </div>
  </div>
</Fragment>
`;
